src/main/java/com/example/bookstoreapp/servive/impl/BookServiceImpl.java

```diff
@@ -1,24 +1,38 @@
 package com.example.bookstoreapp.servive.impl;
 
+import com.example.bookstoreapp.dto.BookDto;
+import com.example.bookstoreapp.dto.CreateBookRequestDto;
+import com.example.bookstoreapp.mapper.BookMapper;
 import com.example.bookstoreapp.model.Book;
 import com.example.bookstoreapp.repository.BookRepository;
 import com.example.bookstoreapp.servive.BookService;
+import jakarta.persistence.EntityNotFoundException;
 import java.util.List;
 import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.stereotype.Service;
 
 @Service
 public class BookServiceImpl implements BookService {
+    private BookMapper bookMapper;
     @Autowired
     private BookRepository bookRepository;
 
     @Override
-    public Book save(Book book) {
-        return bookRepository.save(book);
+    public BookDto save(CreateBookRequestDto createBookRequestDto) {
+        return bookMapper.bookToBookDto(bookRepository.save(
+                bookMapper.createBookRequestDtoToBook(createBookRequestDto)));
     }
 
     @Override
-    public List<Book> findAll() {
-        return bookRepository.findAll();
+    public List<BookDto> findAll() {
+        return bookRepository.findAll().stream()
+                .map(bookMapper::bookToBookDto)
+                .toList();
+    }
+
+    public BookDto getBookById(Long id) {
+        Book book = bookRepository.findById(id)
+                .orElseThrow(() -> new EntityNotFoundException("Book not found with id: " + id));
+        return bookMapper.bookToBookDto(book);
```

Comment:
a small but pretty suggestion
```suggestion
        return bookRepository.findById(id)
                .map(bookMapper::bookToBookDto)
                .orElseThrow(() -> new EntityNotFoundException("Book not found with id: " + id));
```